### ISSUE TITLE
Controlled input is not working properly

### ISSUE THREAD

**Issue by andrebnassis** [tbjgolden/react-codemirror6#2]
Hello, 

I just tried to use your library and it looks like your controlled component is not working properly. Here it is what is happening:

If you do not set onChange, the component still updates the text, so, it doesn't matter what you are filling on 'value' or 'onChange' props. Check the code below:

```typescript
import React from 'react'
import { CodeMirror } from 'react-codemirror6'

const App = () => {
 
  return (
    <div style={{ height: 400, display: 'flex' }}>
      <CodeMirror
        style={{
          display: 'flex',
          flexDirection: 'column',
          flex: '1 0 auto'
        }}
        value={'this text should not change itself when typing'}
        onChange={(value:string) => console.log(value)}
        extensions={
          [
            // theme, language, ...
          ]
        }
      />
    </div>
  )
}

export default App;
```
The sample code that I provide you should keep the text 'this text should not change itself when typing' on the editor, no matter what you type. Unfortunately, it is being changed when you type.

Do you have an idea of how to fix it?

**Comment by tmgldn** [comment]
Thanks @andrebnassis

It's been a long time since I wrote this component so I'm surprised that this isn't working; will take a look now!

**Comment by tbjgolden** [comment]
Okay so looking at the code, the issue is because the current approach is to replace the content of the editor with any changed values passed in as props, but it doesn't work in the same way as a traditional controlled input.

I'll look over the code and see if I can find a new approach that makes that possible!

The main issues I'm seeing are:
- how to persist things like multiple cursors, and
- how not interrupt the typing experience

**Comment by tmgldn** [comment]
@andrebnassis 

Fixed in v1.0.1 dafe16a16ad19277184b037c4e8dc9e9037913fe

**Comment by andrebnassis** [comment]
Sorry for the late response, I just tested and it works perfectly! Thanks a lot! :D
